### ISSUE TITLE
FletcherChecksum8() may generate invalid VRID

### ISSUE THREAD

**Issue by mandre** [openshift/baremetal-runtimecfg#21]
Sometimes, keepalived pod may end up in a crash loop with the following logs.

```
Starting Keepalived v1.3.5 (03/19,2017), git commit v1.3.5-6-g6fa32f2
Opening file '/etc/keepalived/keepalived.conf'.
Starting VRRP child process, pid=7
Registering Kernel netlink reflector
Registering Kernel netlink command channel
Registering gratuitous ARP shared channel
Opening file '/etc/keepalived/keepalived.conf'.
Truncating auth_pass to 8 characters
VRRP Error : VRID not valid - must be between 1 & 255. reconfigure !
Truncating auth_pass to 8 characters
VRRP_Instance(l3vvqqni-f21ca_DNS) the virtual id must be set!
Stopped
Keepalived_vrrp exited with permanent error CONFIG. Terminating
Stopping
Stopped Keepalived v1.3.5 (03/19,2017), git commit v1.3.5-6-g6fa32f2
```

It turns out, the `FletcherChecksum8()` function used to compute a unique VRID may return `0` with certain input such as "l3vvqqni-f21ca-dns.
`0` is not a valid VRID.

https://play.golang.org/p/V0i11g1XvsY

**Comment by cybertron** [comment]
Hmm, apparently the keepalived docs lie: "arbitrary unique number from 0 to 255" (from the virtual_router_id section of https://www.keepalived.org/manpage.html).

Also, comparing our fletcher8 implementation to some online calculators I'm getting different results for a given input value. I think to make ours "correct", we need to do:

-               ckA = (ckA + inp[i]) & 0xf
-               ckB = (ckB + ckA) & 0xf
+               ckA = (ckA + inp[i]) % 0xf
+               ckB = (ckB + ckA) % 0xf

However, for our purposes I don't think it matters. Both implementations give us an 8 bit value that changes depending on the input.

**Comment by mandre** [comment]
I believe this has been fixed with #44 
/close
